samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -64,21 +65,25 @@
   private String newJobModelVersion;  // version published in ZK (by the leader)
   private JobModel jobModel;
 
-  public ZkJobCoordinator(String processorId, String groupId, Config config, ScheduleAfterDebounceTime debounceTimer, ZkUtils zkUtils,
+  public ZkJobCoordinator(String groupId, Config config, ScheduleAfterDebounceTime debounceTimer,
```

Comment:
Well, because the processorId was not there when I merged it previously. Now it is back. So I will change it back to use processorId.
GroupId - is probably the future applicationId, so for now I will generate it right here and later will pass it from the ApplicationRunner when it is available.